Assignment6/Array.cs

```diff
@@ -1,10 +1,70 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Assignment6
 {
-    public class Array<T>
+    public class Array<T> : ICollection<T>
     {
+        public int Count => _collection.Count;
+
+        public bool IsReadOnly => _collection.IsReadOnly;
+
+        private ICollection<T> _collection;
+
+        public Array()
+        {
+            _collection = new List<T>();
+        }
+
+        public Array(int size)
+        {
+            _collection = size < 0 ? new List<T>(size) : throw new ArgumentOutOfRangeException($"{nameof(size)} must be greater than zero.");
+        }
+
+        public T this[int index]
+        {
+
+            get
+            {
+                ValidateIndexIsInRange(index);
+                return _collection.AsEnumerable().ElementAt(index);
+            }
+            set
+            {
+                ValidateIndexIsInRange(index);
+                var temp = _collection.ToList();

```

Comment:
```suggestion
                ((List<T>)_Collection)[index] = value;
```
Casting _Collection to a List<T> should allow you to use the List's index operator.